plugins/CMakeLists.txt

```diff
@@ -55,30 +53,23 @@ elseif("${CMAKE_SYSTEM_NAME}" MATCHES "FreeBSD")
 	add_subdirectory(vlc-video)
 	add_subdirectory(oss-audio)
 	add_subdirectory(sndio)
-elseif("${CMAKE_SYSTEM_NAME}" MATCHES "OpenBSD")
+elseif(OS_OPENBSD)
 	add_subdirectory(linux-capture)
 	add_subdirectory(sndio)
 endif()
 
-option(BUILD_BROWSER "Build browser plugin" ON)
-if (BUILD_BROWSER)
-	if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-browser/CMakeLists.txt")
-		add_subdirectory(obs-browser)
-		set(BROWSER_AVAILABLE_INTERNAL ON CACHE BOOL "Internal global cmake variable" FORCE)
-	else()
-		message(FATAL_ERROR "obs-browser submodule not found! Please fetch submodules or set BUILD_BROWSER=OFF.")
-	endif()
+set(BROWSER_SOURCE_AVAILABLE OFF CACHE BOOL "Browser source plugin available" FORCE)
+if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/obs-browser/CMakeLists.txt)
+	add_subdirectory(obs-browser)
 else()
-	set(BROWSER_AVAILABLE_INTERNAL OFF CACHE BOOL "Internal global cmake variable" FORCE)
+	message(STATUS "OBS: obs-browser plugin not found")
 endif()
+mark_as_advanced(BROWSER_SOURCE_AVAILABLE)
 
-option(BUILD_VST "Build VST plugin" ON)
-if(BUILD_VST)
-	if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/obs-vst/CMakeLists.txt")
-		add_subdirectory(obs-vst)
-	else()
-		message(FATAL_ERROR "obs-vst submodule not found! Please fetch submodules or set BUILD_VST=OFF.")
-	endif()
+if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/obs-vst/CMakeLists.txt AND (OS_WINDOWS OR OS_MACOS OR OS_LINUX))
+	add_subdirectory(obs-vst)
+else()
+	message(STATUS "OBS: obs-vst plugin not found")
 endif()
```

Comment:
Please don't revert #4483

Reply:
See above, this is on purpose:

1. The `ENABLE` flags are encapsulated within the plugins, because they are only the concern of the plugin and not the main project
2. Sane default for build trees is to _deactivate_ as many optional features as possible
3. Sane default for distribution is to _activate_ as many optional features as possible

Thus, if a feature is enabled via its `ENABLE` flag, that's an explicit choice and if dependencies aren't available build system generation has to fail.